data/core/doc/init.lua

```diff
@@ -499,6 +544,34 @@ function Doc:indent_text(unindent, line1, col1, line2, col2)
   return line1, col1 + #text, line1, col1 + #text
 end
 
+function Doc:update_max_line_len_range(start_line, end_line)
+  local line_numbers = self.long_lines.line_numbers
+  local max_length = self.long_lines.length
+  end_line = math.min(end_line, #self.lines)
+
+  for line=start_line,end_line do
```

Comment:
We may use the same approach I suggested above to add items in the line and remove the spurious entries at the end. We have the problem that the value for each key entry is "true" and not equal to its len. Actually we may leave the values in the table with the length of the line instead of using "true"s values.